src/libponyrt/sched/scheduler.c

```diff
@@ -727,6 +742,22 @@ static pony_actor_t* steal(scheduler_t* sched)
       if (actor != NULL)
         break;
     }
+
+    // if we're scheduler 0 and cycle detection is enabled
+    if((sched->index == 0) && !ponyint_actor_getnoblock())
```

Comment:
does the value of `ponyint_actor_getnoblock` change during the course of a run? if not, these cases should probably be flipped.